source/server/http/admin.cc

```diff
@@ -328,14 +353,13 @@ void AdminImpl::writeClustersAsJson(Buffer::Instance& response) {
 
         envoy::admin::v2alpha::HostHealthStatus& health_status =
             *host_status.mutable_health_status();
-        health_status.set_failed_active_health_check(
```

Comment:
I'm confused. Instead of all the enum macro magic, couldn't you just switch on the original enum here and achieve the same thing? The compiler would error if there is a case that is not handled as long as there is no default? I'm probably missing something.

Reply:
Essentially the issue is that any number of enum flags can be set at the same time, so we need to check each one

Reply:
Sorry yes, this makes sense. This is a neat trick for making sure to cover every enum value. I can't think of a better way to do this.